agera/src/main/java/com/google/android/agera/CompiledRepository.java

```diff
@@ -225,13 +231,20 @@ private boolean checkCancellationLocked() {
    */
   void acknowledgeCancel() {
     boolean shouldStartFlow = false;
+    Object discardedIntermediateValue = null;
     synchronized (this) {
       if (runState == CANCEL_REQUESTED) {
         runState = IDLE;
+        if (intermediateValue != currentValue && intermediateValue != initialValue) {
```

Comment:
I removed the initial value check instead. Added clarifications in the javadoc.

There are further improvements to be made too: initialValue doesn't need to be kept if there's no RESET_TO_INITIAL_VALUE config. This allows the initial value to be GC'd and justifies the disposer treatment a bit better. But this is very minor, I think, so doesn't need to be dealt with in this PR.